solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -250,16 +250,18 @@ private HttpClient createHttpClient(Builder builder) {
   public void close() {
     // we wait for async requests, so far devs don't want to give sugar for this
     asyncTracker.waitForComplete();
-    if (closeClient) {
-      try {
+    try {
+      if (closeClient) {
         httpClient.setStopTimeout(1000);
         httpClient.stop();
-      } catch (Exception e) {
-        throw new RuntimeException("Exception on closing client", e);
+        httpClient.destroy();
```

Comment:
I have no idea if `destroy()` is needed, but I was surprised it existed. When I stepped through with the debugger, `destroy` cleaned up more resources which seems to be a good idea on `close()`...

Reply:
https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/client/HttpClient.html is `Destroyable` (https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/util/component/Destroyable.html)

> Typically a Destroyable is a [LifeCycle](https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/util/component/LifeCycle.html) component that can hold onto resources over multiple start/stop cycles. A call to destroy will release all resources and will prevent any further start/stop cycles from being successful.

So I think we actually want to destroy the `httpClient` here since we can't start the `HttpClient` again anyway...

Reply:
https://github.com/apache/solr/blob/main/solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java#L242 is the only place we call `HttpClient#start()` so we should call stop and destroy when we are done.

Reply:
Makes sense to call destroy to me.